clang/labs/lab1/polozov_vladislav/WarningForDeprecated.cpp

```diff
@@ -0,0 +1,60 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "llvm/Support/raw_ostream.h"
+#include <algorithm>
+using namespace clang;
+
+class WarningDeprecatedConsumer : public ASTConsumer {
+  CompilerInstance &Instance;
+
+public:
+  WarningDeprecatedConsumer(CompilerInstance &Instance) : Instance(Instance) {}
+
+  void HandleTranslationUnit(ASTContext &context) override {
+
+    struct Visitor : public RecursiveASTVisitor<Visitor> {
+      ASTContext *Context;
+      Visitor(ASTContext *_Context) : Context(_Context) {}
+      bool VisitFunctionDecl(FunctionDecl *FD) {
+        std::string want_find = "deprecated";
+        std::string name = FD->getNameInfo().getAsString();
+        std::transform(name.begin(),name.end(),name.begin(), ::tolower);
```

Comment:
Why it ignores case?

Reply:
> Why it ignores case?

Can you please elaborate on what exactly to fix?
I got the following comment "Also could you please add option for case insensitive check of deprecated?" . I thought that I should make the check so that the case of the word "deprecated" is not taken into account. That's why I translate the whole function name into lower case and already look for the occurrence of "deprecated"

Reply:
@aobolensk Yes, it was my modification suggestion for the task.

Reply:
OK